### PR TITLE
requirejs_main tag

### PR DESCRIPTION
**Pull request by millerdev**
@orangejenny

### CODE REVIEW

File: corehq/apps/hqwebapp/templatetags/tests/templates/requirejs_base.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{% load hq_shared_tags %}
+{% if requirejs_main %}
+unexpected {{ requirejs_main }} before block
+{% endif %}
+
+{% requirejs_main "requirejs/base" %}
+{% block requirejs_main %}{% endblock %}
```
Comment (millerdev) [inline corehq/apps/hqwebapp/templatetags/tests/templates/requirejs_base.html]: oops, this line should be removed (it has no effect and is not needed)

File: corehq/apps/hqwebapp/templatetags/hq_shared_tags.py [hunk 2]
```diff
@@ -674,14 +673,54 @@ def render(self, context):
...
+
+
+class RequireJSMainNode(template.Node):
+
+    def __init__(self, name, value):
+        self.name = name
+        self.value = value
+
+    def __repr__(self):
+        return "<RequireJSMain Node: %r>" % (self.value,)
```
Comment (orangejenny) [inline corehq/apps/hqwebapp/templatetags/hq_shared_tags.py]: is the trailing comma intentional?

Comment (millerdev) [reply corehq/apps/hqwebapp/templatetags/hq_shared_tags.py]: Yes, it's a workaround for a common pitfall with `%` string formatting in Python. The comma makes it a single-item tuple containing `self.value`, which guarantees that the string formatting expression will always succeed (even if `self.value` is a multi-item tuple, for example, although I guess that's not much of a concern here).

Comment (orangejenny) [reply corehq/apps/hqwebapp/templatetags/hq_shared_tags.py]: aha, interesting

**Review by orangejenny — APPROVED** [review]

**Pull request merged**